src/ui/pages/equipment/position/PositionHierarchy.js

```diff
@@ -3,9 +3,67 @@ import EISPanel from 'eam-components/dist/ui/components/panel';
 import EAMInput from 'eam-components/dist/ui/components/muiinputs/EAMInput'
 import EAMAutocomplete from 'eam-components/dist/ui/components/muiinputs/EAMAutocomplete'
 import WSEquipment from "../../../../tools/WSEquipment";
+import EAMCheckbox from 'eam-components/dist/ui/components/muiinputs/EAMCheckbox';
 
 class PositionHierarchy extends Component {
 
+    styles = {
+        checkboxStyle: {
+            display:"flex", 
+            paddingLeft: "10px",
+            marginRight:"-23px",
+            marginBottom: "-20px",
+            marginTop: "-10px"
+        },
+        labelStyle: {
+            display:"flex", 
+            alignItems:"center",
+            padding:'0',
+            marginBottom: "-20px",
+            marginTop: "-10px"
+        },
+        dependencyRowStyle: {
+            display: "flex", 
+            flexDirection:"row-reverse", 
+            fontSize:"10px", 
+            width: '100%'
+        },
+        dependencyCheckboxWrapperStyle: {
+            display: "flex", 
+            flexDirection:"row",
+        }
+    }
+
+    updateDependencyProperty = (key, value, equipment) => {
+        let { updateEquipmentProperty } = this.props
+        //inversed map of hierarchy properties. Asset Dependency has the highest priority.
+        
+        let hierarchyMap = {
+            "hierarchyPrimarySystemCode": "hierarchyPrimarySystemDependent",
+            "hierarchyPositionCode": "hierarchyPositionDependent",
```

Comment:
Asset?

Reply:
there is no asset dependency in position